flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -616,10 +394,9 @@ public static long calculateHeapSizeMB(long totalJavaMemorySizeMB, Configuration
 		Preconditions.checkArgument(totalJavaMemorySizeMB > 0);
 
 		// subtract the Java memory used for network buffers (always off-heap)
-		final long networkBufMB =
-			calculateNetworkBufferMemory(
-				totalJavaMemorySizeMB << 20, // megabytes to bytes
-				config) >> 20; // bytes to megabytes
+		final long networkBufMB = NetworkEnvironmentConfiguration.calculateNetworkBufferMemory(
```

Comment:
It looks like a future API point for ShuffleService to estimate the memory it needs in container.

Reply:
Yes, it might be. Then we further focus on it future?

Reply:
I think so, I created FLINK-12110 to track it.